### ISSUE TITLE
[Bug]: docs-page table-of-contents does not scroll and highlight current paragraph, if not enough height available

### ISSUE THREAD

**Issue by kroeder** [storybookjs/storybook#26347]
### Describe the bug

In docs page, if you click on a table-of-contents item, and there is not enough height available in the docs page, it actually does not change the current state of the table-of-contents.

![table-of-contents](https://github.com/storybookjs/storybook/assets/5478899/a54972a2-206e-4e3a-9813-9c04269159b3)

At work, I dealt with a similar issue (horizontal tabs with scrollable content section). What we did was to use a hidden element that adds as many height as needed to be able to scroll to even the last element in all cases. It produces lots of unused space at the bottom but from a UX perspective, you can now use the toc and scroll + show the current active toc item in all cases.

Specifically, we use a resizeObserver to check if the user browser height changes. If it does, we calculate and apply the needed additional height.

### To Reproduce

[ttps://stackblitz.com/edit/github-w91kck?file=.storybook%2Fpreview.ts](https://stackblitz.com/edit/github-w91kck?file=.storybook%2Fpreview.ts)

- Open docs page
- Click on an item and see the same result as in the video capture above

### Additional context

**Alternative solution;** Do not add a hidden container at the bottom, leave everything at is but at least underline the last clicked "toc" element. After user-scroll, this obviously would change again but is still less confusing IMO

**Comment by vanessayuenn** [comment]
> At work, I dealt with a similar issue (horizontal tabs with scrollable content section). What we did was to use a hidden element that adds as many height as needed to be able to scroll to even the last element in all cases. It produces lots of unused space at the bottom but from a UX perspective, you can now use the toc and scroll + show the current active toc item in all cases.
>
> Specifically, we use a resizeObserver to check if the user browser height changes. If it does, we calculate and apply the needed additional height.

This seems like a good approach! However, I don't think the team will have capacity to implement this in near future. But PRs are most certainly welcome! ✨ 

For anyone interested in taking this on: be sure to check the [contribution guide](https://storybook.js.org/docs/react/contribute/how-to-contribute) and hop onto the [Storybook Discord](https://discord.gg/storybook) if you need any help getting started.

**Comment by maudem** [comment]
@vanessayuenn Do you mind if I take this?

**Comment by vanessayuenn** [comment]
Please do @maudem! Be sure to check the [contribution guide](https://storybook.js.org/docs/react/contribute/how-to-contribute) and hop onto the [Storybook Discord](https://discord.gg/storybook) if you need any help getting started. Looking forward to your contribution! ✨

**Comment by Sidnioulz** [comment]
Thinking about fixing this bug? Please see the comments on #26470 especially https://github.com/storybookjs/storybook/pull/26470#issuecomment-2560817209.

**Comment by idiglove** [comment]
I have done some digging on this issue.

TLDR: The problem is in the `tocbot` package. It has been resolved by the maintainer here - https://github.com/tscanlin/tocbot/pull/370.

I tested it locally and we can see that it works and the link clicked is highlighted.
https://github.com/user-attachments/assets/5a7cce1f-e28b-4bfd-bbe0-71ef309073fa

The fix is in tocbot@4.33.0, in this case I'm using higher than that.

![Image](https://github.com/user-attachments/assets/bfd24859-d7b4-4095-9273-7887d44b8462)

We can close this issue :)

**Comment by Sidnioulz** [comment]
Thanks @idiglove, closing!
